compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -396,34 +398,53 @@ impl<'a, 'tcx> AbstractConstBuilder<'a, 'tcx> {
             | ExprKind::Field { .. }
             | ExprKind::ConstBlock { .. }
             | ExprKind::Adt(_) => self.error(
+                Some(node.span),
+                "unsupported operation in generic constant, this may be supported in the future",
+            )?,
+            ExprKind::Pointer { .. } =>
+            // dont know if this is correct
+            {
+                self.error(Some(node.span), "pointers are not allowed in generic constants")?
+            }
+            ExprKind::Continue { .. } | ExprKind::Break { .. } | ExprKind::Loop { .. } => self
+                .error(
                     Some(node.span),
-                    "unsupported operation in generic constant, this may be supported in the future",
+                    "loops and loop control flow are not supported in generic constants",
                 )?,
+            ExprKind::Box { .. } =>
+            // allocations not allowed in constants
+            {
+                self.error(Some(node.span), "allocations are not allowed in generic constants")?
+            }
+
+            // we handle valid unary/binary ops above
+            ExprKind::Unary { .. } | ExprKind::Binary { .. } | ExprKind::LogicalOp { .. } => {
+                self.error(Some(node.span), "unsupported operation in generic constants")?
+            }
+            ExprKind::Assign { .. } | ExprKind::AssignOp { .. } => {
+                self.error(Some(node.span), "assignment is not supported in generic constants")?
+            }
+            ExprKind::Closure { .. } | ExprKind::Return { .. } => self.error(
+                Some(node.span),
+                "closures and function keywords are not supported in generic constants",
+            )?,
+            ExprKind::Match { .. } | ExprKind::If { .. } | ExprKind::Let { .. } =>
```

Comment:
I think we should handle `Yield` `Return` `Continue` `Break` `Loop` in this arm as the reason they are all disallowed is because we dont allow controlflow